lib/app/services/push_notification_service.dart

```diff
@@ -0,0 +1,184 @@
+import 'dart:convert';
+
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:get_it/get_it.dart';
+import 'package:gruene_app/app/constants/secure_storage_keys.dart';
+import 'package:gruene_app/app/enums/push_notification_topic_enum.dart';
+import 'package:gruene_app/app/models/push_notification_settings_model.dart';
+import 'package:gruene_app/app/utils/logger.dart';
+import 'package:jwt_decoder/jwt_decoder.dart';
+
+class PushNotificationService {
+  late PushNotificationSettingsModel _settings;
+  final FlutterSecureStorage _secureStorage = GetIt.instance<FlutterSecureStorage>();
+
+  Future<void> initialize() async {
+    await Firebase.initializeApp();
+
+    final fb = FirebaseMessaging.instance;
+    await fb.requestPermission();
+
+    try {
+      await fb.requestPermission();
+    } catch (e) {
+      logger.e('Failed to initialize Firebase Messaging: $e');
+    }
+
+    _settings = await _loadSettings();
+  }
+
+  /// Load Push Notification settings from secure storage
+  PushNotificationSettingsModel getSettings() {
+    return _settings;
+  }
+
+  /// Load Push Notification settings from secure storage
+  Future<PushNotificationSettingsModel> _loadSettings() async {
+    // load enabled flag from storage
+    final enabledVal = await _secureStorage.read(key: SecureStorageKeys.pnEnabled);
+    final enabled = enabledVal != null ? jsonDecode(enabledVal) as bool : true;
```

Comment:
🔧 I think this would be way more descriptive/readable. Same below.
```suggestion
    final enabledJson = await _secureStorage.read(key: SecureStorageKeys.pnEnabled);
    final enabled = enabledJson != null ? jsonDecode(enabledJson) as bool : true;
```